Library/Homebrew/tap.rb

```diff
@@ -630,3 +622,27 @@ def []=(key, value)
     value
   end
 end
+
+class Manpages
+  def self.install(path, command)
+    link_src_dst_dirs(path/"man", HOMEBREW_PREFIX/"share/man", command)
+  end
+
+  def self.uninstall(path)
+    unlink_src_dst_dirs(path/"man", HOMEBREW_PREFIX/"share/man")
+  end
+end
+
+class Completions
+  def self.install(path, command)
+    link_src_dst_dirs(path/"completions/bash", HOMEBREW_PREFIX/"etc/bash_completion.d", command)
+    link_src_dst_dirs(path/"completions/zsh", HOMEBREW_PREFIX/"share/zsh/site-functions", command)
+    link_src_dst_dirs(path/"completions/fish", HOMEBREW_PREFIX/"share/fish/vendor_completions.d", command)
+  end
+
+  def self.uninstall(path)
+    unlink_src_dst_dirs(path/"completions/bash", HOMEBREW_PREFIX/"etc/bash_completion.d")
+    unlink_src_dst_dirs(path/"completions/zsh", HOMEBREW_PREFIX/"share/zsh/site-functions")
+    unlink_src_dst_dirs(path/"completions/fish", HOMEBREW_PREFIX/"share/fish/vendor_completions.d")
+  end
```

Comment:
Agreed. I wasn't quite sure of Homebrew's OOP vs Procedural code conventions here and kind of expected that this might be your take.